packages/puppeteer-core/src/common/Input.ts

```diff
@@ -538,6 +538,15 @@ export class Mouse {
   constructor(client: CDPSession, keyboard: Keyboard) {
     this.#client = client;
     this.#keyboard = keyboard;
+
+    // Reset the state whenever the frame navigates.
+    client.on('Page.frameNavigated', () => {
+      this.#_state = {
+        position: {x: 0, y: 0},
+        buttons: MouseButtonFlag.None,
+      };
+      this.#transactions = [];
```

Comment:
what if there are scheduled transactions in there? is there any other situation when the positions have to be reset.